src/gltf/GltfModelObjectMetaData.hpp

```diff
@@ -0,0 +1,139 @@
+*  derived from this software without specific prior written permission from the respective party.
+*
+*  (4) Other than as required in clauses (1) and (2), distributions in any form of modifications or other derivative works
+*  may not use the "OpenStudio" trademark, "OS", "os", or any other confusingly similar designation without specific prior
+*  written permission from Alliance for Sustainable Energy, LLC.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+#ifndef GLTF_GLTFMODELOBJECTMETADATA_HPP
+#define GLTF_GLTFMODELOBJECTMETADATA_HPP
+
+#include "GltfAPI.hpp"
+
+#include "../utilities/core/Logger.hpp"
+
+namespace tinygltf {
+class Value;
+}
+
+namespace openstudio {
+namespace model {
+
+  class AirLoopHVAC;
+  class BuildingUnit;
+  class SpaceType;
+  class BuildingStory;
+  class ThermalZone;
+  class DefaultConstructionSet;
+  class Space;
+}  // namespace model
+
+namespace gltf {
+
+  class GltfMetaData;
+
+  /** GltfModelObjectMetaData is an Interface class between select ModelObjects and glTF values
+   *
+   *  It is capable of storing attributes such as name, iddObjectType, color, handle from ModelObjects:
+   *  ThermalZone, SpaceType, AirLoopHVAC, BuildingStory, BuildingUnit, DefaultConstructionSet
+   **/
+  class GLTF_API GltfModelObjectMetaData
+  {
+   public:
+    /** Default constructor **/
+    GltfModelObjectMetaData();
+
+    /** @name Interface with ModelObjects */
+    //@{
+    explicit GltfModelObjectMetaData(const model::AirLoopHVAC& airLoopHVAC);
+    explicit GltfModelObjectMetaData(const model::BuildingUnit& buildingUnit);
+    explicit GltfModelObjectMetaData(const model::SpaceType& spaceType);
+    explicit GltfModelObjectMetaData(const model::BuildingStory& buildingStory);
+    explicit GltfModelObjectMetaData(const model::ThermalZone& thermalZone);
+    explicit GltfModelObjectMetaData(const model::DefaultConstructionSet& defaultConstructionSet);
+    explicit GltfModelObjectMetaData(const model::Space& space);
+    //@}
```

Comment:
I won't be commenting every class, but I'll do this one as an example.

As you can see, this has several constructor that take in a derived ModelObject to populate the interface members